src/libraries/Microsoft.Extensions.DependencyModel/tests/Microsoft.Extensions.DependencyModel.Tests.csproj

```diff
@@ -9,6 +9,9 @@
 
     <!-- FluentAssertions library has a generic recursion and the warning breaks NativeAOT build -->
     <NoWarn>$(NoWarn);IL3054</NoWarn>
+
+    <!-- Microsoft.DotNet.ProjectModel brings in NS1.x dependencies -->
```

Comment:
Curious, what is Microsoft.DotNet.ProjectModel? Is that a library that we don't produce live anymore?

Reply:
yeah it is some old library from project.json days I think: https://www.nuget.org/packages/Microsoft.DotNet.ProjectModel/

I'm not sure the tests that use it provide much value so we might consider deleting the code.

Reply:
This source file specifically: https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.DependencyModel/tests/DependencyContextBuilderTests.cs

Reply:
@ericstj had a quick chat about this. The `DependencyContextBuilder` type that is used in those tests lives in that `Microsoft.DotNet.ProjectModel` package but at some point then got moved into dotnet/sdk: https://github.com/dotnet/sdk/blob/bedb8ccb29ebda43103d0f1559106db704e15b52/src/Tasks/Microsoft.NET.Build.Tasks/DependencyContextBuilder.cs#L14

SDK doesn't reference the ProjectModel package at all and it's already marked as deprecated on nuget.org: https://www.nuget.org/packages/Microsoft.DotNet.ProjectModel/1.0.0-rc3-1-003177#versions-body-tab

I'm skeptical that we still need the tests but if we do, we should move them into dotnet/sdk next to the existing ones: https://github.com/dotnet/sdk/blob/bedb8ccb29ebda43103d0f1559106db704e15b52/src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenADependencyContextBuilder.cs. That said, I don't think that's necessary. They should definitely be deleted from here though as the ProjectModel package 

cc @eerhardt who moved DependencyModel into runtime: https://github.com/dotnet/runtime/commit/b93829cf38f079ccb895708bb7e938ab9541bcca

Reply:
See comment in https://github.com/dotnet/runtime/pull/34296#discussion_r401855280. Copied here:

> Both Microsoft.DotNet.InternalAbstractions and Microsoft.DotNet.ProjectModel are obsolete packages (think project.json and .xproj time frame). The issue is our unit tests are written relying on these old packages, and we never took the time to rewrite the tests. But we don't want to lose the test coverage.

Reply:
I'm not so sure on that - this single file `DependencyContextBuilderTests.cs` is the only one using those old references.  AFAICT every test here is just calling `DependencyContextBuilder.Build` and making sure the results are as expected.  That's mostly testing `ProjectModel` and not `DependencyModel`.  Sure `DependencyContextBuilder` returns types defined in `DependencyModel` but I'm not sure if we're getting much out of these tests that we don't get out of the unit tests for those types and APICompat.